### PR TITLE
Add UIControl extension for simply adding an action

### PR DESCRIPTION
**Pull request by SimonNumberTwo**
# Description
This merge request adds a simple way to add an action to a UIControl when the primary action was triggered.

### CODE REVIEW

File: Sources/Extension/UIControl+extension.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//  Copyright © 2020 Jamit Labs GmbH. All rights reserved.
+
+import UIKit
+
+@objc
+class ClosureWrapper: NSObject {
```
Comment (mrylmz) [inline Sources/Extension/UIControl+extension.swift:6]: Please make this class private

File: Sources/Extension/UIControl+extension.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+}
+
+public extension UIControl {
+    struct AssociatedKey {
+        static var actionWrapper: UInt8 = 0
+    }
+
+    /// Add an action to a UIControl by simply passing the control event. The closure will then be notified when this event is being triggered.
+    func addAction(for controlEvent: UIControl.Event = .primaryActionTriggered, _ closure: VoidCallback?) {
```
Comment (mrylmz) [inline Sources/Extension/UIControl+extension.swift:25]: Please rename the parameter closure to `callback`

File: Sources/Extension/UIControl+extension.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    @objc
+    func invoke () {
+        closure?()
+    }
+}
+
+public extension UIControl {
+    struct AssociatedKey {
```
Comment (mrylmz) [inline Sources/Extension/UIControl+extension.swift:20]: Please make this struct private

**Review by mrylmz — CHANGES_REQUESTED** [review]

File: Sources/Extension/UIControl+extension.swift (hunk 2 above)
Comment (SimonNumberTwo) [reply Sources/Extension/UIControl+extension.swift:25]: Done

File: Sources/Extension/UIControl+extension.swift (hunk 3 above)
Comment (SimonNumberTwo) [reply Sources/Extension/UIControl+extension.swift:20]: Done.

**Pull request merged**